app/controllers/users_controller.rb

```diff
@@ -1,37 +1,78 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, except: [:new, :create]
+  before_action :find_user, except: [:index, :new, :create]
+  before_action :correct_user, only: [:edit, :update]
+
   def new
+    redirect_to(root_path) if logged_in?
     @user = User.new
   end
 
   def index
     @users = User.all
   end
 
-  def show
-    @user = User.find_by id: params[:id]
-    return if @user
+  def show; end
 
-    flash[:danger] = t "users.index.not_found"
-    redirect_to root_path
-  end
+  def edit; end
 
   def create
     @user = User.new user_params
-
     if @user.save
-      log_in @user
-      flash[:success] = t "users.show.welcome", name: @user.name
-      redirect_to @user
+      @user.send_activation_email
+      flash[:info] = t "users.create.info"
+      redirect_to login_url
     else
-      show_errors_messages @user.errors.messages
+      flash[:danger] = t "users.create.fail"
       render :new
     end
   end
 
+  def update
+    if @user.update user_params
+      flash[:success] = t "users.update.success"
+      redirect_to @user
+    else
+      flash[:danger] = t "users.update.fail"
+      render :edit
+    end
+  end
+
+  def destroy
+    if current_user.admin
+      if @user != current_user && @user.destroy
```

Comment:
```suggestion
      if current_user.admin && @user != current_user && @user.destroy
```
Xét chung như thế này được không em

Reply:
Được ạ